File: P1047/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main () {
    int roadL;
    int m;
    int tree = 0;
    cin >> roadL >> m;
    vector<int> startN(m);
    vector<int> finN(m);
    vector<int> treeN(roadL + 1);

    for (int i = 0; i < m; i++){
        cin >> startN.at(i) >> finN.at(i) ;
    }
    
    for (int i = 0; i < (roadL +1); i++){
        treeN.at(i) = 1;
    }

    for (int i = 0; i < m; i++){
        for (; startN.at(i) <= finN.at(i); startN.at(i)++){
            treeN.at(startN.at(i)) = 0;   
        }
    }

    for (int i =0; i< (roadL + 1); i++){
        tree += treeN.at(i);

    }

    cout << tree;
    return 0;
}

File: P1047/main.py
roadL, m = map(int,input().split())
tree = 0
startN = [None] * m
finN = [None] * m
treeN = [None] * (roadL + 1)

for i in range (m):
    startN[i], finN[i] = map(int,input().split())

for i in range  (roadL + 1):
    treeN[i] = 1

for i in range (m):
    while startN[i] <= finN[i]:
        treeN[startN[i]] = 0
        startN[i] += 1

for i in range (roadL + 1):
    tree += treeN[i]

print(tree)
